gateway/ln-gateway/src/lib.rs

```diff
@@ -920,6 +920,30 @@ impl Gateway {
         ))
     }
 
+    async fn handle_create_invoice_for_self_msg(
+        &self,
+        payload: CreateInvoiceForSelfPayload,
+    ) -> Result<Bolt11Invoice> {
+        let GatewayState::Running { lightning_context } = self.get_state().await else {
+            return Err(GatewayError::Disconnected);
+        };
+
+        Bolt11Invoice::from_str(
+            &lightning_context
+                .lnrpc
+                .create_invoice(CreateInvoiceRequest {
+                    payment_hash: Vec::new(), /* Empty payment hash indicates an invoice payable
```

Comment:
Can we make this more explicit? At least an `Option<Vec<u8>>`?

Reply:
Nope 🙁 `CreateInvoiceRequest` is a proto message, so the [default value](https://protobuf.dev/programming-guides/proto3/#default) for bytes is an empty list of bytes.